certbot/certbot/_internal/main.py

```diff
@@ -1394,6 +1394,18 @@ def run(config: configuration.NamespaceConfig,
     except errors.PluginSelectionError as e:
         return str(e)
 
+    if config.must_staple and installer and "staple-ocsp" not in installer.supported_enhancements():
+        raise errors.NotSupportedError("Must-Staple extension requested, but OCSP stapling "
+                                       "is not supported by the selected installer "
+                                       f"({config.installer})\n\nYou can either:\n * remove the "
+                                       "`--must-staple` option from the command line and obtain a "
+                                       "certificate without the Must-Staple extension, or;\n * use "
+                                       "the `certonly` subcommand and manually install the "
+                                       "certificate into your webserver. Note that OCSP stapling "
```

Comment:
Not sure about "webserver" as I'm reading the text again.. Other services can use TLS certs too of course, although I'm not sure if there are other services besides webservers/webclients actually implementing Must-Staple..

Reply:
Replaced "webserver" with "intended service (e.g. webserver)".